### DISCUSSION TITLE
Windows update broke my winget. New installer does not install.

### DISCUSSION THREAD

**Discussion by Bartolomeus383** [microsoft/winget-cli#4983] [Q&A]
I have a problem with winget that presumably occurred after the installation of KB5046613 (dated November 13). 

_EDIT: I'm sorry, the format in this post is hardly readable, and I don't know how to fix it. In edit mode all is well. 
EDIT2: Thanks Florelis for the edit, my post is now readable. Can't I paste text from the powershell directly?_
```
**_Program 'winget.exe' failed to run: Het systeem kan geen toegang verkrijgen tot het bestandAt line:1 char:1**
+ winget -v
+ ~~~~~~~~~.
At line:1 char:1
+ winget -v
+ ~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed_
```

**Translation of the first line of the output in Dutch:**
**System cannot access the fileAt**

I attempted to install the latest version of Winget. Output in popup:

**_[path to installer and installer name]
The installation failed. Please contact the software vendor_**

I noticed that the msxbundle was much larger than the version I downloaded almost 2 years ago (ment for a machine that did not have Winget installed natively). The new one from last week is about 231MB, the old one from Feb 2023 only 15MB. Does an explanation of the problem lie in this difference in size? If not, can someone explain in a few sentences why the msxbundle got so much bigger?

What do I need to do to be able to use Winget again?

**Comment by jshergal** [comment]
I have the same problem and it occurred after the last update in Windows 10. I tried downloading the app bundle again and it fails to run. I tried updating through windows store and that also failed.

```powershell
ResourceUnavailable: Program 'winget.exe' failed to run: An error occurred trying to start process 'C:\Users\xxx\AppData\Local\Microsoft\WindowsApps\winget.exe'
```

**Comment by Bartolomeus383** [comment]
1
I looked in the location:
C:username]AppDataLocalApps
and found that my winget.exe has a size of 0kb
Is your winget.exe also 'sizeless', @jshergal?

2
I don't use the Windows Store. I looked anyway and found winget NOT, only the winget related:
Widgets for UniGetUI (formerly WingetUI).
Winget-AutoUpdate-aaS
UniGetUI
Why don't I see winget itself in the windows store?

**Reply by Masamune3210** [reply]
1. Winget is not located in AppData\Local, that is just a shortcut so that the system can launch the program without having to specify the path.
2. Winget is part of another package, namly, the App Installer package, Microsoft.DesktopAppInstaller

**Reply by jshergal** [reply]
Hi @Bartolomeus383,
Yes at that location the size of winget.exe on my system is also 0. There are apparently 2 versions installed on my system currently, but I receive an access denied error when trying to run them:

C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.24.25200.0_x64__8wekyb3d8bbwe\winget.exe
C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.24.25180.0_x64__8wekyb3d8bbwe\winget.exe

I could not find "App Installer" as it is called, when searching the Windows Store. I was able to find it in the Windows Store when I navigated there from my "Apps/Programs" section of settings. I have tried updating, removing, etc. and all I get is "an error occurred, we are aware of this and working on it" or something to that effect. Even running UniGetUI which has its own version of Winget packaged with it seems to fail to actually run winget, so there is something strange that happened as a result of the latest windows update. 
I have tried `sfc /scannow` and it said everything was fine. Maybe it is related to me uninstalling Copilot, which magically appeared in my taskbar after the latest update. I removed it from the taskbar and then attempted to uninstall from the apps/programs area of settings. When I did that it worked a while, then grayed out the icon for copilot in the menu and said "an error occurred". Not sure if it is related, but maybe there was some copilot cruft added to winget.

**Comment by jshergal** [comment]
I found a fix, the issue is related to a WinAppSDK update. I found I was having an issue with other Windows Store apps and so in searching for a fix for that, I came across this post:

https://github.com/microsoft/WindowsAppSDK/issues/4881#issuecomment-2485701825

I followed the steps listed in that comment and winget is now working again, along with other Windows Store apps.

**Reply by Bartolomeus383** [reply]
Thanks @jshergal for getting back to me. I do run Windows build 19045. The post you link to states 'There is an upcoming Windows Update which will fix this issue'. I hope that comes true, for the time being I'll get by with my other software updaters then.

I'll get back myself if things don't work out after the next Windows update.
